### PR TITLE
Allow non-Prior parameters in Prior and media transformations

### PR DESCRIPTION
**Pull request by wd60622**
## Description

```python
import numpy as np
import pymc as pm

from pymc_marketing.mmm import GeometricAdstock

adstock = GeometricAdstock(l_max=12, priors={"alpha": 1.0})

X = np.random.randn(100)

with pm.Model() as model:
    adstocked_X = adstock.apply(X)
```
Some plotting will not be supported if there are no distribution parameters required

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pymc_marketing/mmm/components/base.py [hunk 1]
```diff
@@ -117,7 +120,9 @@ class Transformation:
     lookup_name: str
 
     def __init__(
-        self, priors: dict[str, Prior] | None = None, prefix: str | None = None
+        self,
+        priors: Prior | float | TensorVariable | None = None,
```
Comment (juanitorduz) [inline pymc_marketing/mmm/components/base.py:124]: Can you please update the type hints in the doctrines 🙏 ?

File: pymc_marketing/mmm/components/base.py [hunk 2]
```diff
@@ -556,3 +573,26 @@ def apply(self, x: pt.TensorLike, dims: Dims | None = None) -> TensorVariable:
         """
         kwargs = self._create_distributions(dims=dims)
         return self.function(x, **kwargs)
+
+
+def _serialize_value(value: Any) -> Any:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/components/base.py:578]: Can you please add a test for this function :) ?

**Review by juanitorduz — CHANGES_REQUESTED** [review]
Looks nice! Two small requests :) WDYT?

File: pymc_marketing/mmm/components/base.py (hunk 1 above)
Comment (wd60622) [reply pymc_marketing/mmm/components/base.py]: Done!

File: pymc_marketing/mmm/components/base.py (hunk 2 above)
Comment (wd60622) [reply pymc_marketing/mmm/components/base.py:581]: Done. All serializations are tested (through various tests.)

File: pymc_marketing/mmm/components/base.py [hunk 3]
```diff
@@ -556,3 +576,26 @@ def apply(self, x: pt.TensorLike, dims: Dims | None = None) -> TensorVariable:
...
+    if isinstance(value, TensorVariable):
+        value = value.eval()
+
+    if isinstance(value, np.ndarray):
+        return value.tolist()
+
+    return value
+
+
+def _deserialize(value):
```
Comment (wd60622) [inline pymc_marketing/mmm/components/base.py:597]: This will be expanded upon in deserialization PR

**Review by juanitorduz — APPROVED** [review]
Than you @wd60622 !

**Pull request merged**